### DISCUSSION TITLE
Trouble with Optix in CMakeLists.txt

### DISCUSSION THREAD

**Discussion by Phlakdal** [NVlabs/instant-ngp#560] [Q&A]
Hi ! I have an error while running the following command "cmake . -B build".

CMake Warning at CMakeLists.txt:163 (find_package):
  By not providing "FindOptiX.cmake" in CMAKE_MODULE_PATH this project has
  asked CMake to find a package configuration file provided by "OptiX", but
  CMake did not find one.

  Could not find a package configuration file provided by "OptiX" with any of
  the following names:

    OptiXConfig.cmake
    optix-config.cmake

  Add the installation prefix of "OptiX" to CMAKE_PREFIX_PATH or set
  "OptiX_DIR" to a directory containing one of the above files.  If "OptiX"
  provides a separate development package or SDK, be sure it has been
  installed.

CMake Warning at CMakeLists.txt:170 (message):
  OptiX was not found.  Neural graphics primitives will still compile and run
  correctly.  However, SDF training in 'raystab' and 'pathescape' modes will
  be significantly slower.

**Comment by satyajit-ink** [comment]
set the optix install dir environment variable

```
OptiX_INSTALL_DIR=/usr/local/NVIDIA-OptiX-SDK-7.4.0-linux64-x86_64  cmake ..
```

**Comment by Phlakdal** [comment]
Thank you, I had already tried with "OptiX_DIR" and tried again but either my path is wrong or it is not the solution.
![image](https://user-images.githubusercontent.com/62513146/165951003-20022a70-23d5-4c89-a05f-5f25800cb90a.png)

**Comment by satyajit-ink** [comment]
You path should be `C:\ProgramData\NVIDIA Corporation\OptiX SDK 7.4.0\`

well atleast for linux it worked to set it as root directory of OptiX, should work for windows as ll.

**Comment by Phlakdal** [comment]
I fixed this problem a few days ago by uninstalling visual studio 2022 and installing visual studio 2019.

**Reply by nowhereman1999** [reply]
Hi,I meet this problem when I try to build the project,the cmake also can not find my OptiX even when I set the path in the CMakeLists.txt
![image](https://user-images.githubusercontent.com/53631206/171778629-043d0cbb-e446-4c26-9502-5c7d09f7eed8.png)
![image](https://user-images.githubusercontent.com/53631206/171778915-604fc6ed-5860-4424-bebe-57238b0652de.png)
However,the version of my Visual Studio is already 2019,so how can I fix it and could you give me some advice?

**Reply by Phlakdal** [reply]
Hi, as i said i had to uninstall VS 2022 too, so if you have an other version installed it could be the cause.

If not I remember that my first solution was to look up for optix and install it, don’t know if you did this, for me it didn’t work until after i corrected my VS version but maybe this was part of the fix.

**Reply by nowhereman1999** [reply]
Thanks for your sincere reply and I have solved it by adding the path of the root directory of OptiX to my system environment variable，but when I execute the command "cmake --build build --config RelWithDebInfo -j 16" to generate the project,it throw a lot of errors like this:
![image](https://user-images.githubusercontent.com/53631206/171827615-262f1fe4-0361-499d-80ff-f120ff566c32.png)
so I search the error in the issues again and found [https://github.com/NVlabs/instant-ngp/issues/178](url) which says it can be worked out by install the cuda 11.6 ,However,the current cuda installed on my computer is 11.3 and I checked highest cuda version supported by my RTX2060 is 11.4 which is shown in the figure below:
![image](https://user-images.githubusercontent.com/53631206/171828785-5731647e-7daf-4e52-8e12-648a2478bd70.png)
So I even can't try to install a new cuda 11.6 to run the command again and find out wether it works,have you meet the same problem with me and could you please give me some advice again？:-)

**Reply by Phlakdal** [reply]
Have you tried with Cuda 11.4 then ? And did you add CUDA to your path ? I haven’t had this issue but i can’t try to help.

**Reply by nowhereman1999** [reply]
> Have you tried with Cuda 11.4 then ? I haven’t had this issue but i can’t try to help.

Thank you for your prompt reply again,I haven't try with cuda 11.4 because there are many past projects of mine rely on the cuda 11.3 ,and I have add CUDA to my path too because the nvcc -V works well：
![image](https://user-images.githubusercontent.com/53631206/171831340-0a664b7a-c81d-4f9e-b65d-17bc1ee29f79.png)
so now I want to install multiple cuda version on my laptop and try it again which is a little bit inconvenient:-(

**Reply by Phlakdal** [reply]
You might not have to. Apparently you can get more information on the error :

![image](https://user-images.githubusercontent.com/62513146/171834913-fdf8df66-80c0-42c1-b531-b0030acc9a58.png)

https://forums.developer.nvidia.com/t/how-to-fix-error-msb3721/47471

**Reply by nowhereman1999** [reply]
OK,I will try to find more informations through this error

**Reply by nowhereman1999** [reply]
I‘m so glad to tell you that I have successfully compiled the project after I install the CUDAv11.6,it seems that this is really due to the conflict between the nvcc of CUDAv11.3 and the MSVC of VS2019:
![image](https://user-images.githubusercontent.com/53631206/172034658-1f2feeea-b438-4621-bf33-8d526b33f7e5.png)

**Reply by Phlakdal** [reply]
Interesting, well done !
